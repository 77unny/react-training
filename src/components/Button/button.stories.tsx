import React from "react";
import { Story, Meta } from "@storybook/react";
import Button, { IButtonProps } from "./index";
import "../../styles/styles.css";

export default {
  title: "Components/Button",
  component: Button,
} as Meta;

const Template: Story<IButtonProps> = (args) => <Button {...args} />;

export const Default = Template.bind({});
Default.args = {
  label: "Button",
};

export const AllButton = () => (
  <div>
    <div style={{ margin: "5px 0" }}>
      <Template label="기본버튼" />
      <Template label="클릭버튼" onClick={() => console.log("Click!")} />
    </div>
  </div>
);
