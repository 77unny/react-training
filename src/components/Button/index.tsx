import React, { FC } from "react";
import styled from "styled-components";
import tw from "tailwind-styled-components";

const ContainerStyled = styled.button`
  &:hover {
    font-weight: bold;
  }
`;
const Container = tw(ContainerStyled)`custom-button`;

export interface IButtonProps {
  label: string;
  onClick?: () => void;
}

const Button: FC<IButtonProps> = ({ label, onClick }) => {
  return <Container onClick={onClick}>{label}</Container>;
};

export default Button;
