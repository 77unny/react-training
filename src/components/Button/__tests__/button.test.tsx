import React from "react";
import { render, screen } from "@testing-library/react";
import Button from "../index";

describe("<Button />", () => {
  it("Button 컴포넌트 렌더링", () => {
    const { container } = render(<Button label="버튼" />);
    expect(container).toMatchSnapshot();
    const label = screen.getByText("버튼");
    expect(label).toBeInTheDocument();
  });
});
