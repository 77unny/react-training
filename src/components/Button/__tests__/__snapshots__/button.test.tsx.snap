// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button 컴포넌트 렌더링 1`] = `
<div>
  <button
    class="sc-bdnxRM kVRuzd custom-button"
  >
    버튼
  </button>
</div>
`;
